### PR TITLE
PR: Override RichJupyterWidget._get_color() because we use a custom style (IPython console)

### LINKED ISSUE #3798: Invert colors for "Pretty-Printing" symbolic math in IPython's dark background
The pretty printed symbolic math is unreadable when the the IPython console background is set to dark.

To reproduce:
1. Set preferences -> IPython console->Display->Background color to dark
2. Set preferences -> IPython console->Advanced->Use symbolic math
3. Restart ipython console
4. Enter a numeric command into Ipython and execute

Expected: Visible output
Got: only readable output when selected with the mouse to invert colors

* Spyder Version: 3.0.2
* Python Version: 3.5.2 64bit
* Operating system: Linux (Plasma5)

jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.2 (OK)
pandas >=0.13.1  :  0.19.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
psutil >=0.3     :  5.0.0 (OK)
pyflakes >=0.6.0 :  1.3.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5 (OK)
sympy >=0.7.3    :  1.0 (OK)

### LINKED ISSUE #14895: Latex rendering with dark theme is broken
When I use Astropy quantities, they are rendered as black-text latex in Spyder's ipython console. This is fine when the console has a light background, but unreadable when I choose a dark theme for Spyder. There is a similar issue with pint.
I was not able to find settings anywhere in Astropy/ipython/Spyder to fix this.
Astropy defines a _repr_latex_ method. The only workaround I could find was to delete the method.

### PR DESCRIPTION
**Pull request by bnavigator**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

This is a follow-up on https://github.com/jupyter/qtconsole/pull/512

| before | This PR + https://github.com/jupyter/qtconsole/pull/512 + ipython/ipython#13372 |
| --- | --- |
| image | image |

### Issue(s) Resolved

Fixes #14895
Fixes last comments (non-sympy) in #3798

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: bnavigator

### CODE REVIEW

File: spyder/plugins/ipythonconsole/widgets/shell.py [hunk 1]
```diff
@@ -945,6 +946,17 @@ def _syntax_style_changed(self):
         else:
             self._highlighter.set_style_sheet(self.style_sheet)
 
+    def _get_colors(self, color):
+        """
+        Get a color as qtconsole.styles.get_colors() would return from
+        a builtin Pygments style
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/shell.py:952]:
```suggestion
        a builtin Pygments style.
```

Missing period.

**Review by ccordoba12 — COMMENTED** [review]
Thanks a lot for your help with this @bnavigator! It's great to have the right foreground color for equations for any library and not just Sympy.

Besides the tiny comment I left for you below, please also remove the calls to `set_sympy_forecolor` shown below because they are not necessary anymore.

https://github.com/spyder-ide/spyder/blob/a43307653e738e35f8c600d5b4e13857c4b80f03/spyder/plugins/ipythonconsole/widgets/shell.py#L326-L332

**Comment by bnavigator** [comment]
> please also remove the calls to `set_sympy_forecolor` shown below because they are not necessary anymore.

Are you sure about this? Sympy uses its own set of printers and is not affected by the change for the QtConsole/IPython renderer of `_repr_latex_()` attributes.

image

With your suggested change:
image

**Comment by ccordoba12** [comment]
> Are you sure about this?

I thought that call wasn't necessary anymore. Thanks a lot for checking that out!

**Comment by oscargus** [comment]
Note that SymPy behaves differently before and after calling `init_printing()`. Before calling `init_printing` the actual image is rendered by IPython (similar to the left hand side in the original figure), while after it is rendered by SymPy.

`"%colors lightbg"` is the flexible way to tell SymPy what a good guess of foreground colors are when using its own rendering, see https://ipython.readthedocs.io/en/stable/config/details.html#terminal-colors

File: spyder/plugins/ipythonconsole/widgets/shell.py [hunk 2]
```diff
@@ -945,6 +946,17 @@ def _syntax_style_changed(self):
         else:
             self._highlighter.set_style_sheet(self.style_sheet)
 
+    def _get_colors(self, color):
+        """
+        Get a color as qtconsole.styles.get_colors() would return from
+        a builtin Pygments style.
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/shell.py:952]:
```suggestion
    def _get_color(self, color):
        """
        Get a color as qtconsole.styles._get_color() would return from
        a builtin Pygments style.
```

**Review by ccordoba12 — COMMENTED** [review]
@bnavigator, some small suggestions to go with the ones I made in PR jupyter/qtconsole#512. The this should be ready.

**Review by ccordoba12 — APPROVED** [review]
@bnavigator, thanks for your help in all related repos to solve this issue. It's a terrific improvement!

Note: The failure in our tests is unrelated to this change.

**Pull request merged**